modules/Bio/EnsEMBL/Compara/RunnableDB/PairAligner/AlignmentChains.pm

```diff
@@ -162,54 +148,51 @@ sub fetch_input {
     }
   }
   
+  $self->param('features', $features);
   print STDERR scalar @{$features}," features at time: ",scalar(localtime),"\n";
 
-  my %parameters = (-analysis             => $fake_analysis,
-                    -features             => $features,
-                    -workdir              => $self->worker_temp_directory,
-		    -linear_gap           => $self->param('linear_gap'));
-  
   $self->compara_dba->dbc->disconnect_if_idle();
   # Let's keep the number of connections / disconnections to the minimum
   $qy_gdb->db_adaptor->dbc->prevent_disconnect( sub {
-      my $query_slice = $self->param('query_dnafrag')->slice;
       my $query_nib_dir = $self->param('query_nib_dir');
-      $parameters{'-query_slice'} = $query_slice;
       # If there is no .nib file, preload the sequence
-      if ($query_nib_dir and (-d $query_nib_dir) and (-e $query_nib_dir . "/" . $query_slice->seq_region_name . ".nib")) {
-          print STDERR "reusing the query nib file\n";
-          $parameters{'-query_nib_dir'} = $query_nib_dir;
+      if ($query_nib_dir and (-d $query_nib_dir) and (-e $query_nib_dir . "/" . $self->param('query_dnafrag')->name . ".nib")) {
+          print STDERR "reusing the query nib file ". $query_nib_dir . "/" . $self->param('query_dnafrag')->name . ".nib" . "\n";
       } else {
           print STDERR "fetching the query sequence\n";
+          my $query_slice = $self->param('query_dnafrag')->slice;
+          $self->param('query_slice', $query_slice);
```

Comment:
Why do we only want to store the query_slice as a param if we're not reusing a .nib file?

Reply:
Found the answer in Bio::EnsEMBL::Compara::Production::Analysis::AlignmentChains

Reply:
Nice. I didn't remember and was digging back into the code :)